VideoWeb/VideoWeb/ClientApp/src/app/shared/select-media-devices/select-media-devices.component.ts

```diff
@@ -44,30 +45,7 @@ export class SelectMediaDevicesComponent implements OnInit, OnDestroy, AfterView
     ) {}
 
     ngAfterViewInit() {
-        // create a trap focus for the modal window
-        const element = document.getElementById('select-device-modal');
-        const focusableEls = element.querySelectorAll(
-            'a[href]:not([disabled]), button:not([disabled]), textarea:not([disabled]), input[type="text"]:not([disabled]), input[type="radio"]:not([disabled]), input[type="checkbox"]:not([disabled]), select:not([disabled])'
-        );
-
-        const firstFocusableEl = focusableEls[0];
-        const lastFocusableEl = focusableEls[focusableEls.length - 1];
-
-        const KEYCODE_TAB = 9;
-
-        element.addEventListener('keydown', function (e) {
-            if (e.key === 'Tab' || e.keyCode === KEYCODE_TAB) {
-                if (e.shiftKey) {
-                    /* shift + tab */ if (document.activeElement === firstFocusableEl) {
-                        e.preventDefault();
-                    }
-                } /* tab */ else {
-                    if (document.activeElement === lastFocusableEl) {
-                        e.preventDefault();
-                    }
-                }
-            }
-        });
+        ModalTrapFocus.trap('select-device-modal');
```

Comment:
Could we use a ViewChild here instead, so that we can re-use this element later if needed?

`@ViewChild('selectDeviceModal') selectDeviceModal: ElementRef;`

There's an existing example in that component.